docs/html/user_guide.rst

```diff
@@ -861,6 +861,296 @@ of ability. Some examples that you could consider include:
+      cachetools==4.2.4 --hash sha256:92971d3cb7d2a97efff7c7bb1657f21a8f5fb309a37530537c71b1774189f2d1
+      certifi==2021.10.8 --hash sha256:d62a0163eb4c2344ac042ab2bdf75399a71a2d8c7d47eac2e2ee91b9d6339569
+      charset-normalizer==2.0.10 --hash sha256:cb957888737fc0bbcd78e3df769addb41fd1ff8cf950dc9e7ad7793f1bf44455
+      idna==3.3 --hash sha256:84d9dd047ffa80596e0f246e2eab0b391788b0503584e8945f2368256d2735ff
+      pyasn1-modules==0.2.8 --hash sha256:a50b808ffeb97cb3601dd25981f6b016cbb3d31fbf57a8b8a87428e6158d0c74
+      requests-oauthlib==1.3.0 --hash sha256:7f71572defaecd16372f9006f33c2ec8c077c3cfa6f5911a9a90202beb513f3d
+      rsa==4.8 --hash sha256:95c5d300c4e879ee69708c428ba566c59478fd653cc3a22243eeb8ed846950bb
+      six==1.16.0 --hash sha256:8abb2f1d86890a2dfb989f9a77cfcfd3e47c2a354b01111771326f8aa26e0254
+      urllib3==1.26.7 --hash sha256:c4fdf4019605b6e5423637e01bc9fe4daef873709a7973e195ceba0a62bbc844
+      oauthlib==3.1.1 --hash sha256:42bf6354c2ed8c6acb54d971fce6f88193d97297e18602a3a886603f9d7730cc
+      pyasn1==0.4.8 --hash sha256:39c7e2ec30515947ff4e87fb6f456dfc6e84857d34be479c9d4a4ba4bf46aa5d
+      setuptools==60.3.1 --hash sha256:2932bfeb248c648dc411ea9714d5a6de7a33ef1a0db2f0fce644d8172b0479e8
+
+JSON schema
+-----------
+
+The JSON report is described below with fields given fake mypy type annotations corresponding to the output of `json.load() <https://docs.python.org/3/library/json.html#json.load>`_ upon reading the JSON report. The report contains multiple top-level fields:
+
+* ``experimental: bool``: set to ``True`` as the format is not yet stable.
+* ``input_requirements: List[str]``: strings describing the requirements provided to the pip resolver. Can be parsed with `packaging.requirements.Requirement <https://packaging.pypa.io/en/latest/requirements.html>`_.
+* ``python_version: str``: a string describing the python interpreter version the resolve was performed for. Can be parsed with `packaging.specifiers.SpecifierSet <https://packaging.pypa.io/en/latest/specifiers.html>`_. Currently always an exact ``==`` constraint.
+* ``candidates: Dict``: each package that would have been downloaded with ``pip download`` is represented in the ``candidates`` dict. Each key is the ``name`` of a requirement, since each dependency package name is satisfied by exactly one candidate in the final resolve.
+
+Candidates
+----------
+
+Each element of the ``candidates`` dict has the following fields:
+
+* ``requirement: str``: an ``==`` requirement string for the exact version of the candidate that would have been fetched by ``pip download``. Can be parsed with `packaging.requirements.Requirement <https://packaging.pypa.io/en/latest/requirements.html>`_.
+* ``requires_python: Optional[str]``: a constraint on the executing python interpreter version exerted by this candidate. Can be parsed with `packaging.specifiers.SpecifierSet <https://packaging.pypa.io/en/latest/specifiers.html>`_.
```

Comment:
Here I'd suggest a `plaform` field, providing a `python_version` field, which can later be extended with additional information such as platform tags, etc.

Reply:
Totally agree, otherwise a user may consume a resolution report for the wrong platform.

Reply:
To elaborate, I think these should be the env_vars from [PEP 508](https://peps.python.org/pep-0508/#grammar) which then allow the evaluation of environment markers using the `packaging` library.